### ISSUE TITLE
Prettier Strips Parens from JSDoc Type Cast

### ISSUE THREAD

**Issue by rbiggs** [prettier/prettier#4710]
**Prettier 1.13.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgBAARgTwAc4tgAJAFQFkAZAUQBs4BbBGAXy3VSwAo4mrWAEoAdACcWEAG5wAgjBjiAlgCMArvF6FxEQsJAAaEHpjLoAZ2SgAhuN0B3AAp2EVlDekRlAEyMhVcRswAGs4GABlQmDlKABzZCV1OGNYizhxGCcguOYbZAAzGwZ04wArCwAPACEg0PCIm1YaWLhC4tKQaPF08WQAm1V8Bmh-HViYAHVfGAALZAAOAAZjHQh0yaDCfp04Xtl-SQBHdWVJbJtc-KQikpSQdOZlRPFk4wtYuKYARXUIeHad2MMEG0x8c2QACZgUFlAxPgBhCDMPL9KDQNrGdTpCiDdy3dLsdhAA)
```sh
--parser babylon
```

**Input:**
```jsx
/** @type {HTMLElement} */ (element).removeAttribute(prop)
```

**Output:**
```jsx
/** @type {HTMLElement} */ element.removeAttribute(prop);

```

**Expected behavior:**

```jsx
/** @type {HTMLElement} */ (element).removeAttribute(prop)
```

JSDoc supports type casting by using an inline comment with the type and by enclosing the target element in parens. Prettier is stripping those parens out, causing the type cast to fail. In the above JavaScript, `element` is of type `Node`, which only has prositional properties and methods. To use `setAttribute`, `removeAttribute` and other Element features, you need to type cast the element to type HTMLElement. This is done with the inline comment and by enclosing the object to cast in parens. Removing those parens causes the type checker to flag the element as an error since `removeAttribute` does not exist on type Node. 

You can read more about JSDoc type casting here (scroll down to the bottom of the page): https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler
Or here in the Visual Studio Code documentation: https://github.com/Microsoft/TypeScript/wiki/JSDoc-support-in-JavaScript. Scroll down to the last item in the supported features, right before `Patterns that are known NOT to be supported`

**Comment by ikatyang** [comment]
Hi, this looks like a duplicate of #4287 and should be fixed by #4648.

**Prettier pr-4648**
[Playground link](https://deploy-preview-4648--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6AVOgBAARgTwAc4tgAJAFQFkAZAUQBs4BbBGAXy3VSwAo4mrWAEoAdACcWEAG5wAgjBjiAlgCMArvF6FxEQsJAAaEHpjLoAZ2SgAhuN0B3AAp2EVlDekRlAEyMhVcRswAGs4GABlQmDlKABzZCV1OGNYizhxGCcguOYbZAAzGwZ04wArCwAPACEg0PCIm1YaWLhC4tKQaPF08WQAm1V8Bmh-HViYAHVfGAALZAAOAAZjHQh0yaDCfp04Xtl-SQBHdWVJbJtc-KQikpSQdOZlRPFk4wtYuKYARXUIeHad2MMEG0x8c2QACZgUFlAxPgBhCDMPL9KDQNrGdTpCiDdy3dLsdhAA)
```sh
--parser babylon
```

**Input:**
```jsx
/** @type {HTMLElement} */ (element).removeAttribute(prop)
```

**Output:**
```jsx
/** @type {HTMLElement} */ (element).removeAttribute(prop);

```

**Comment by ikatyang** [comment]
Duplicate of #4287
